Explain RPC behavior when using Tensor as arg or return value

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31968 Explain RPC behavior when using Tensor as arg or return value**

Differential Revision: [D19321380](https://our.internmc.facebook.com/intern/diff/D19321380)

mrshenli committed:
Explain RPC behavior when using Tensor as arg or return value

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit e5fc7ba8:


**None of the build failures appear to be your fault**.

* **1/1** [broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=d2fdf140affc443f7df9a52023306826c75de49a&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35) at merge base d2fdf140 since Jan 08
  <details>
  <summary>
  You may want to rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=e5fc7ba8f4b16b92e5121b51ba92148bf2bd694c).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably **caused by upstream breakages**:
* [pytorch_macos_10_13_py3_test](https://dr.pytorch.org/build-details.html?build_id=82331364) from Jan 08



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

rohan-varma reviewed (approved):
LGTM, pinging @jjlilley to ensure that this is indeed the current behavior

mrshenli committed:
Update on "Explain RPC behavior when using Tensor as arg or return value"


Differential Revision: [D19321380](https://our.internmc.facebook.com/intern/diff/D19321380)

[ghstack-poisoned]
